### PR TITLE
Remove twisted dependency from component API

### PR DESCRIPTION
**Pull request by om26er**
Fix for https://github.com/crossbario/autobahn-python/issues/907

### CODE REVIEW

**Comment by meejah** [comment]
This looks fine as-is .. but I wonder if we could move the "Twisted-specific" code to `autobahn.twisted.component` instead, somehow?

**Comment by om26er** [comment]
> This looks fine as-is .. but I wonder if we could move the "Twisted-specific" code to autobahn.twisted.component instead, somehow?

Now made `done_callback` as a parameter which both backends must provide, kindly take a look again.

File: autobahn/asyncio/component.py [hunk 1]
```diff
@@ -469,8 +469,11 @@ def nicely_exit(signal):
     loop.add_signal_handler(signal.SIGINT, partial(nicely_exit, 'SIGINT'))
     loop.add_signal_handler(signal.SIGTERM, partial(nicely_exit, 'SIGTERM'))
 
+    def done_callback(reactor, arg):
+        reactor.stop()
+
     # returns a future; could run_until_complete() but see below
```
Comment (meejah) [inline autobahn/asyncio/component.py]: very minor nit-pick (and I'm not even sure if we're consistent here already) but asyncio usually calls the reactor "loop"

**Comment by meejah** [comment]
Great! This looks nicer to me; keeping any Twisted/asyncio-specific imports in `asyncio.twisted.*` is much better than inside `if` constructs :)

**Comment by meejah** [comment]
(If you want to squash it all and force-push, go ahead .. otherwise good to merge from my perspective).

File: autobahn/asyncio/component.py (hunk 1 above)
Comment (om26er) [reply autobahn/asyncio/component.py]: just pushed another commit that changes that.

**Pull request merged**